internal/database/users.go

```diff
@@ -1129,6 +1147,63 @@ func (u *userStore) getBySQL(ctx context.Context, query *sqlf.Query) ([]*types.U
 	return users, nil
 }
 
+const userForSCIMQueryFmtStr = `
+SELECT u.id,
+       u.username,
+       u.display_name,
+       u.avatar_url,
+       u.created_at,
+       u.updated_at,
+       u.site_admin,
+       u.passwd IS NOT NULL,
+       u.tags,
+       u.invalidated_sessions_at,
+       u.tos_accepted,
+       u.searchable,
+       ARRAY(SELECT email FROM user_emails WHERE user_id = u.id) AS emails,
+       (SELECT account_id FROM user_external_accounts WHERE user_id=u.id AND service_type = 'scim') AS scim_external_id
+  FROM users u %s`
+
+// getBySQLForSCIM returns users matching the SQL query, along with their email addresses and SCIM ExternalID.
+func (u *userStore) getBySQLForSCIM(ctx context.Context, query *sqlf.Query) ([]*types.UserForSCIM, error) {
+	// NOTE: We use a separate query here because we want to fetch the emails and SCIM ExternalID in a single query.
+	q := sqlf.Sprintf(userForSCIMQueryFmtStr, query)
+	rows, err := u.Query(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	// Convert results to users
+	users := []*types.UserForSCIM{}
+	for rows.Next() {
+		u, err := scanUserForSCIM(rows)
+		if err != nil {
+			return nil, err
+		}
+		users = append(users, u)
+	}
+	if err = rows.Err(); err != nil {
+		return nil, err
+	}
+
+	return users, nil
+}
+
```

Comment:
```suggestion
	q := sqlf.Sprintf(userForSCIMQueryFmtStr, query)
	return scanUsersForSCIM(u.Query(ctx, q))
}

var scanUsersForScim = basestore.NewSliceScanner(scanUserForSCIM)
```

Pretty sure this works

Reply:
Wow, this meta-scanner is pretty neat. And it works! Would've taken me a lot of time to find out about it. Will be in my next PR in 2 min